src/tmp/SampleEnginePlugin.ts

```diff
@@ -0,0 +1,176 @@
+import * as EngineApi from '@salesforce/code-analyzer-engine-api';
+
+export class SampleEnginePlugin extends EngineApi.EnginePluginV1 {
```

Comment:
I'm thinking making this engine be part of core as a sample engine might be beneficial: 1) Ravi can take that up work and can be good starter story 2) In the long run, we have a sample engine as an example for any external developers trying to add engine.

Reply:
If we want to put it somewhere, maybe inside of the engine api module at best. But I would be cautious here and not sure if I agree with this idea entirely. Stubs are really for specific test use cases. So I'm not sure if this is the type of example we want universally since its run method returns violations associated with files that don't actually exist.

Reply:
@ravipanguluri - FYI, you could use this as a starting point of what engine plugin and engine would look like.

Reply:
Note that this is just a copy paste of some things from: https://github.com/forcedotcom/code-analyzer-core/blob/dev/packages/code-analyzer-core/test/stubs.ts